### DISCUSSION TITLE
RavenDB nodejs client api return status?

### DISCUSSION THREAD

**Discussion by AnjanaRatnayake** [ravendb/ravendb#15310] [Q&A]
Is there a way to configure DocumentStore (or anything else) to return the status of a client api call.
e.g.
```
let documentstore = new DocumentStore(server_url, database_name);

await documentstore.initialize();

let session = await documentstore.openSession();

session.store(customer);

session.saveChanges();
```
Is there somewhere I can look for the status of saveChanges -- to see if the write has occurred successfully or not?

**Comment by ayende** [comment]
`openSession` is not `async`

https://github.com/ravendb/ravendb-nodejs-client/blob/v5.2/src/Documents/DocumentStore.ts#LL161C4-L161C4

But `store` and `saveChanges` are:

https://github.com/ravendb/ravendb-nodejs-client/blob/v5.2/src/Documents/Session/IDocumentSession.ts#L206
https://github.com/ravendb/ravendb-nodejs-client/blob/v5.2/src/Documents/Session/IDocumentSession.ts#L232

If the `await session.saveChanges()` completed without throwing an exception, the write was successful.

**Reply by AnjanaRatnayake** [reply]
Thanks for your prompt reply!
